Adds KosmicStation

Terra1 opened the pull request:
![image](https://github.com/user-attachments/assets/4f4c8c8b-c921-4ac1-9db2-c042759be1cf)

Adds KosmicStation, a map built based on the "Derelict" ruin and a version of OldStation circa 2012 (with the NorthStar's permabrig added as a "satellite"). It's equipped with **almost** everything a modern station needs, though it's very clearly... quite dated. I didn't have much space to weld things onto without completely destroying the soul, but I believe I did my best.

Here's the Derelict, for comparison:

![image](https://github.com/user-attachments/assets/fc522de6-2805-44e4-85c8-378a42ba9322)

norsvenska commented:
some issues. may have missed some.

<details>
  <summary>these doors are not access locked</summary>

![image](https://github.com/user-attachments/assets/9a163730-6a56-4958-b945-30a98346b620)

</details>

<details>
  <summary>airlock helpers without an airlock</summary>

![image](https://github.com/user-attachments/assets/fb733e94-d46b-47d3-b693-6ad17d8d5efd)

</details>

<details>
  <summary>this bunch of grilles is on normal plating while exposed to space, as well as not being connected to the powernet</summary>

![image](https://github.com/user-attachments/assets/ab68b277-1d36-433c-ba2d-09039fd0826d)

</details>

<details>
  <summary>engineers that leave their department are unable to reenter normally</summary>

![image](https://github.com/user-attachments/assets/498d4bde-21ef-487c-82cd-39e840531b48)

</details>

<details>
  <summary>these should be electrified</summary>

![image](https://github.com/user-attachments/assets/b632fb90-a127-456d-a0ec-aae3ac29c2c9)

</details>

<details>
  <summary>rd's office has no access restrictions or modsuit</summary>

![image](https://github.com/user-attachments/assets/a233dd45-e58d-49e9-9706-8825b381dcf2)

</details>

<details>
  <summary>prison sat is cool but dont just rip northstar's permabrig and slap it there</summary>
  
![image](https://github.com/user-attachments/assets/0c1cc109-5e53-4f58-9308-7bea2c44169e)

![image](https://github.com/user-attachments/assets/6f9c6c96-01af-4552-8034-8da85289ca1a) 
 
</details>


<details>
  <summary>there is no co2 tank and this starts off so it will be clogged</summary>
  
![image](https://github.com/user-attachments/assets/aac857ce-802f-4a44-861c-5e7b9c4e56bb)
  
</details>


<details>
  <summary>biogen and service fab are in walls, also inaccessible to non-chefs and botanists</summary>

![image](https://github.com/user-attachments/assets/7d3e4102-9a3b-4263-8ab9-2ce42d55e875)

</details>


<details>
  <summary>this apc will lose power without intervention. one idea is to rotate it around the wall to connect it to the station grid rather than the solar grid</summary>
  
![image](https://github.com/user-attachments/assets/e5e32b00-ca37-47e3-a994-ae01e85c8ffe)

</details>

Edit: Also, the ordnance bomb site is just nebula's.

Edit 2: there are no radstorm shelters.

Terra1 commented:
> some issues. may have missed some.
> these doors are not access locked

They actually were, but using variables instead of mapping helpers. Fixed that anyway.

> airlock helpers without an airlock

Fixed it. Funny story about that - I used transit tubes for the AI access not only because it's like that on every other map, but on the original OldStation there was an extendable "air bridge" (don't know the exact term) that was accessible by a console, which would provide access. Transit tubes could be considered their spiritual successor of sorts. I would have rather used a tram, but alas, I was short on space and I don't exactly know how to map trams anyway - at least without modifying the game code.

> this bunch of grilles is on normal plating while exposed to space, as well as not being connected to the powernet

Fixed.

> engineers that leave their department are unable to reenter normally

Fixed. The medical lobby now serves a dual-purpose as an engineering lobby.

> these should be electrified

Fixed.

> rd's office has no access restrictions or modsuit

It actually does have a modsuit, but it's buried in the locker as an item. I ran out of space to put the actual suit storage machine. Gave the room access restrictions, though.

> prison sat is cool but dont just rip northstar's permabrig and slap it there

It's supposed to be a joke, in the sense of "CentCom was trying to save money on station construction, so they decided to cut the brig out of a retired station in the scrapyard and weld a solar array to it instead of building something new".

> there is no co2 tank and this starts off so it will be clogged

Fixed it by swapping out the valve for one that starts off on. I would need to fundamentally alter the structure of the station to fit a working CO2 tank in there, and at that point it would no longer be the derelict.

> biogen and service fab are in walls, also inaccessible to non-chefs and botanists

Machinery in the walls is actually pretty common for service areas if you look at other maps. Anyhow, I've fixed the latter issue by making hydroponics accessible to bartenders.

> this apc will lose power without intervention. one idea is to rotate it around the wall to connect it to the station grid rather than the solar grid

Fixed.

> Edit: Also, the ordnance bomb site is just nebula's.

Swapped it out for a mockup of a Syndicate assault pod.

> Edit 2: there are no radstorm shelters.

There is now one radstorm shelter.

Terra1 commented:
I think it's ready now.

EEASAS commented:
is this meant to have no decals or tiles other than normal iron tiles?

EEASAS commented:
![image](https://github.com/user-attachments/assets/7d6d488a-7499-4d17-9c41-f6eb8e6b00c0)
by the way you need an actual cycle thing for these airlocks in sci's burn chamber ordnance thing, look at meta's burn chamber for example

Terra1 commented:
> is this meant to have no decals or tiles other than normal iron tiles?

Yep, it's that nostalgic pre-2012 look I'm going for. Having a bunch of decals might ruin it IMO.

> by the way you need an actual cycle thing for these airlocks in sci's burn chamber ordnance thing, look at meta's burn chamber for example

Thanks, fixed.

Singul0 commented:
very sovl

san7890 commented:
thanks for all the reviews so this works out a bit better for admins who want to run this as an event or something (as I would not have)
